filament/backend/test/BackendTest.h

```diff
@@ -65,24 +66,23 @@ class BackendTest : public ::testing::Test {
             filament::backend::Handle<filament::backend::HwProgram> program,
             const filament::backend::RenderPassParams& params);
 
-    void readPixelsAndAssertHash(const char* testName, size_t width, size_t height,
-            filament::backend::Handle<filament::backend::HwRenderTarget> rt, uint32_t expectedHash,
-            bool exportScreenshot = false);
-
     filament::backend::DriverApi& getDriverApi() { return *commandStream; }
     filament::backend::Driver& getDriver() { return *driver; }
 
+    ImageExpectations& getExpectations() { return *mImageExpectations; }
+
     static bool matchesEnvironment(Backend backend);
     static bool matchesEnvironment(OperatingSystem operatingSystem);
     static bool matchesEnvironment(OperatingSystem operatingSystem, Backend backend);
-
 private:
 
     filament::backend::Driver* driver = nullptr;
     filament::backend::CommandBufferQueue commandBufferQueue;
     std::unique_ptr<filament::backend::DriverApi> commandStream;
 
     filament::backend::Handle<filament::backend::HwBufferObject> uniform;
+
+    std::optional<ImageExpectations> mImageExpectations;
```

Comment:
This is optional to simplify construction order as `initializeDriver` needs to be called before this is constructed.

I can write something gunkier in the constructor, like `mImageExpectations((initializeDriver(), getDriverApi()))` but it obfuscates that work is being done there so I'd prefer this.